### PR TITLE
Fix/6551 error messages and other bugs

### PR DESCRIPTION
**Pull request by nickguendling**
## Description
Fixes several bugs:
- Corona test service error messages have their correct localized descriptions
- Errors on the home screen from loading the test result are shown correctly by loading the test results independently
- The loading state is left on the Home Screen if updating the test result fails
- The Home Screen is representing the current state of registered tests (publishers that are not dispatched to the main queue indicate that a value **will** change, they haven't changed yet!)
- The warn others reminder is only triggered if the test result screen is shown with a positive, not yet submitted test result
- Checkins to submit are reset when entering ExposureSubmissionCheckinsViewController so the background submission doesn't submit them if the user stays on the screen
- Finally we get the original background color back on the Home Screen – white instead of grey. Needs a little creativity in the last footer view to ensure it stays grey even when scrolling.
- Removes the duplication of warn others view controller initialization in the coordinator

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-6525
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-6526
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-6527
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-6550
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-6551
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-6571

### CODE REVIEW

**Review by marcussc — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/Home/HomeTableViewModel.swift [hunk 1]
```diff
@@ -177,13 +187,17 @@ class HomeTableViewModel {
...
 	}
 
-	private func computedRiskAndTestResultsRows(pcrTest: PCRTest? = nil, antigenTest: AntigenTest? = nil) -> [RiskAndTestResultsRow] {
+	private func computedRiskAndTestResultsRows(pcrTest: PCRTest?, antigenTest: AntigenTest?) -> [RiskAndTestResultsRow] {
 		var riskAndTestResultsRows = [RiskAndTestResultsRow]()
-		if !coronaTestService.hasAtLeastOneShownPositiveOrSubmittedTest {
+
+		let hasAtLeastOneShownPositiveOrSubmittedTest = pcrTest?.positiveTestResultWasShown == true || pcrTest?.keysSubmitted == true || antigenTest?.positiveTestResultWasShown == true || antigenTest?.keysSubmitted == true
+
+
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/Home/HomeTableViewModel.swift:195]: extra space?

Comment (nickguendling) [reply src/xcode/ENA/ENA/Source/Scenes/Home/HomeTableViewModel.swift]: 🦅👁️! Removed it 👍

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/TestResult/HomeTestResultCellModel.swift [hunk 2]
```diff
@@ -17,7 +17,36 @@ class HomeTestResultCellModel {
...
 		self.coronaTestService = coronaTestService
+		self.onUpdate = onUpdate
 
+		setup()
+	}
+
+	// MARK: - Internal
+
+	@OpenCombine.Published var title: String! = ""
+	@OpenCombine.Published var subtitle: String?
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/Home/Cells/TestResult/HomeTestResultCellModel.swift:28]: subTitle

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/TestResult/HomeTestResultCellModel.swift [hunk 3]
```diff
@@ -17,7 +17,36 @@ class HomeTestResultCellModel {
...
+	@OpenCombine.Published var description: String! = ""
+	@OpenCombine.Published var footnote: String?
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/Home/Cells/TestResult/HomeTestResultCellModel.swift:30]: footNote

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/TestResult/HomeTestResultCellModel.swift [hunk 4]
```diff
@@ -30,16 +59,20 @@ class HomeTestResultCellModel {
...
 				.store(in: &subscriptions)
 
 			coronaTestService.$pcrTestResultIsLoading
 				.receive(on: DispatchQueue.OCombine(.main))
 				.sink { [weak self] testResultIsLoading in
-					if testResultIsLoading && self?.coronaTestService.pcrTest?.finalTestResultReceivedDate == nil {
-						self?.configureLoading()
-						onUpdate()
+					if self?.coronaTestService.pcrTest?.finalTestResultReceivedDate == nil {
+						if testResultIsLoading {
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/Home/Cells/TestResult/HomeTestResultCellModel.swift:70]: isTestResultLoading ?

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/TestResult/HomeTestResultCellModel.swift (hunk 2 above)
Comment (nickguendling) [reply src/xcode/ENA/ENA/Source/Scenes/Home/Cells/TestResult/HomeTestResultCellModel.swift:28]: Nope, subtitle is one word.

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/TestResult/HomeTestResultCellModel.swift (hunk 3 above)
Comment (nickguendling) [reply src/xcode/ENA/ENA/Source/Scenes/Home/Cells/TestResult/HomeTestResultCellModel.swift:30]: footnote is one word as well.

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/TestResult/HomeTestResultCellModel.swift (hunk 4 above)
Comment (nickguendling) [reply src/xcode/ENA/ENA/Source/Scenes/Home/Cells/TestResult/HomeTestResultCellModel.swift:70]: This way it makes a correct english sentence: "If test result is loading [do x]" I admit being inconsistent with this, but would like to keep it that way.

Comment (naveeddotio) [reply src/xcode/ENA/ENA/Source/Scenes/Home/Cells/TestResult/HomeTestResultCellModel.swift:70]: It's more a personal style and how you prefer, actually I saw isActivityIndicatorHidden in the next line, so in that context, I found it to be little inconsistent.

**Review by naveeddotio — APPROVED** [review]

**Pull request merged**